x509/pkcs8.go

```diff
@@ -103,34 +122,18 @@ func MarshalPKCS8PrivateKey(key interface{}) ([]byte, error) {
 		}
 
 	case ed25519.PrivateKey:
-		privKey.Algo = pkix.AlgorithmIdentifier{Algorithm: OIDPublicKeyEd25519}
-		var err error
-		if privKey.PrivateKey, err = MarshalEd25519PrivateKey(k); err != nil {
-			return nil, fmt.Errorf("x509: failed to marshal Ed25519 private key while building PKCS#8: %v", err)
+		privKey.Algo = pkix.AlgorithmIdentifier{
+			Algorithm: OIDPublicKeyEd25519,
+		}
+		curvePrivateKey, err := asn1.Marshal(k.Seed())
+		if err != nil {
+			return nil, fmt.Errorf("x509: failed to marshal private key: %v", err)
 		}
+		privKey.PrivateKey = curvePrivateKey
 
 	default:
-		return nil, fmt.Errorf("x509: unknown key type while marshalling PKCS#8: %T", key)
+		return nil, fmt.Errorf("x509: unknown key type while marshaling PKCS#8: %T", key)
 	}
 
 	return asn1.Marshal(privKey)
 }
-
-// MarshalEd25519PrivateKey converts an Ed25519 private key to ASN.1 DER encoded form
-// (as an OCTET STRING holding the key seed, as per RFC 8410 section 7).
-func MarshalEd25519PrivateKey(key ed25519.PrivateKey) ([]byte, error) {
```

Comment:
Removing these functions would be a breaking change to the API. It might be less disruptive to keep them, and add a TODO/Issue to remove them before the next major version release.

Reply:
Actually, it looks like those functions were never part of a release, so we can just delete them.